Makefile.genspec

```diff
@@ -1,9 +1,138 @@
+##
+## Changes there to tests, will result in a different set of test files
+## being generated.
+##
+## This Makefile has "SUBDIRS" .. aka Dyninst "Platforms", each containing
+## the test harness generated for that platform.
+##
+## 1) gprolog uses its logic to render test results from a specification.
+##	This is written to 'tuples', and compressed to 'tuples.gz'
+##
+## 2) A set of python scripts reads the tuples rendered by the prolog
+##	engine, and creates CMakefiles, cmake lists, directories, and
+##	source code as appropriate for the test harness.
+##
+##	This is done in a directory hierarchy, which explodes the various
+##	modes of operation of the testsuite, such as pic/non-pic,
+##	static-linked/dynamic-linked, compiler, and a host of other
+##	dyninst variants.
+##	
+##	The "test groups" are also generated, source files which
+##	allow for grouping of tests, which speeds test harness
+##	execution greatly by reducing overhead ... when it works.
 
+## Recommended method to generate a new specification, is to:
+## 1) move the existing platform configurations to a new subdir
+## 2) gunzip -c tuples.gz >tuples   for each OLD platform in that subdir.
+## 3) generate the new specification
+##		make -f Makefile.gespec gen-clean gen-all
+##	nB: You can make 1 platform by specifying it as a target;
+##	when testing to see if anything works at all in the generator,
+##	that is a good place to start!
+## 4) Use both diff and git to compare the results.
+## 5) Test on changed and unchanged platforms using the new config
+##	a) and examine results.log to verify that tests are run
+##	b) see if there are unexpected test differences, such as
+##	   missing tests, regressions in testing, etc
+## 6) Once everything is OK, add/delete files and directories as needed
+##    in git to match the new harness, and commit it.
+##	nB: don't commit the giant 'tuples' files, just the already-in-git
+##		.gz version.. just in case you need to compare the tuples
+##		files for correctness against a prolog invocation on
+## 		a different machine.
+##
+## The first thing to do when wanting to change the test specification
+## is to re-generate the existing specification, using the above process.
+## 1) If you get byte-for-byte results, everything is OK
+## 2) If not, there is likely a prolog issue,
+##    and you need to find a platform which generates byte-for-byte identical
+##    specification before proceeding
+## Use that compatible platform to make the next specification.
+
+## Thse are the DynInst "platforms" (arch+os) that are the tests built for.
+## The order of entries is historical... aka what was here!
+
+SUBDIRS = \
+	i386-unknown-linux2.4		\
+	x86_64-unknown-linux2.4		\
+	ppc32_linux			\
+	ppc64_linux			\
+	i386-unknown-freebsd7.2		\
+	amd64-unknown-freebsd7.2	\
+	ppc64_bgq_ion			\
+	i386-unknown-nt4.0		\
+	aarch64-unknown-linux		\
+
+## BLANK ABOVE
+
+DEPRECATED_SUBDIRS= \
+	arm64-unknown-linux		\
```

Comment:
The only SUBDIRs (i.e., platforms) not deprecated are x86_64-unknown-linux2.4, ppc64_linux, aarch64-unknown-linux. All the others are dead- for testing purposes. We may someday get back around to testing on FreeBSD.